src/realm/group_shared.cpp

```diff
@@ -1233,7 +1241,7 @@ void SharedGroup::do_open(const std::string& path, bool no_create_file, bool is_
         break;
     }
 
-    m_transact_stage = transact_Ready;
+    set_transact_stage(transact_Ready);
```

Comment:
It's likely that somebody will one day assign with `=`. But since it would only break metrics, I'm not sure it's worth the work to avioid it (such as declaring the TransactStage enum as a struct instead and give it overloads, etc)...